### ISSUE TITLE
Support official Apache Kafka images in Kafka Dev Services

### ISSUE THREAD

**Issue by nscuro** [quarkusio/quarkus#43272]
### Description

Dev Services for Kafka currently supports three providers:

* Redpanda (default)
* Strimzi
* Kafka Native

All of which require a slightly different setup:

https://github.com/quarkusio/quarkus/blob/888266f11c87a961fc9ae28c725f71b94fc2b475/extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/DevServicesKafkaProcessor.java#L228-L276

Since Kafka 3.7.0, the Apache foundation finally publishes official container images, in both JVM and native flavor:

* https://hub.docker.com/r/apache/kafka
* https://hub.docker.com/r/apache/kafka-native

The `org.testcontainers:kafka` module already supports them: https://java.testcontainers.org/modules/kafka/#using-orgtestcontainerskafkakafkacontainer

### Implementation ideas

Add an additional `APACHE_KAFKA` provider that defaults to `apache/kafka-native`. Users can override the image to use the JVM-based `apache/kafka` image if desired.

Consider deprecating the `STRIMZI` and `KAFKA_NATIVE` providers.

**Comment by ozangunalp** [comment]
I think we can add an additional dev service provider "Official" for upstream Kafka images. 

FYI In the original kafka-native project, we've planned to support releases until Kafka 4.0 and then deprecate the project. 

We'd keep the kafka-native provider for a while but not update the image.

**Comment by cescoffier** [comment]
Agreed with @ozangunalp. My tests of the official Kafka native are not very convincing. The amount of memory it uses upfront is concerning (in comparison to Ozan's variant).

We can add it as a possibility, but it won't be the default.

**Comment by nscuro** [comment]
That's fair. But for what it's worth, you can limit the heap size via `KAFKA_OPTS` env var:

```shell
docker run -it --rm -e "KAFKA_OPTS=-Xmx256m" apache/kafka-native:3.8.0
```

**Comment by cescoffier** [comment]
Sure, we can limit the memory, but something looks weird - it probably initialize too much or load too many classes. Our kafka-native does not need that.

**Comment by cescoffier** [comment]
@ozangunalp, we may need to move on with this. I'm very worried about the memory issue - in a sense, it may introduce a requirement on the host machine (that needs a lot more memory). So, I would say let's add it as another option, not the default one.

**Comment by sandonjacobs** [comment]
@cescoffier @ozangunalp Do either of you have an open PR or branch you can share regarding this? I'd be glad to take a look.

**Comment by cescoffier** [comment]
I don't believe we have a branch. If you look at the Karl client deployment module you will see the different alternatives we are proposing today.

**Comment by sandonjacobs** [comment]
Could you share a link to "the Karl client deployment module?" Thanks.

**Comment by cescoffier** [comment]
damn, sorry, it's Kafka Client module... Don't ask, it was early.

The code is: https://github.com/quarkusio/quarkus/blob/main/extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaDevServicesBuildTimeConfig.java#L46

The most interesting part for you is: https://github.com/quarkusio/quarkus/blob/main/extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaNativeContainer.java.

**Comment by sandonjacobs** [comment]
I have an implementation I am testing locally that just uses `testcontainers-kafka` and the `org.testcontainers.kafka.KafkaContainer` class. I wanted to start with just the Official Apache Kafka image, per [KIP-1028](https://cwiki.apache.org/confluence/display/KAFKA/KIP-1028%3A+Docker+Official+Image+for+Apache+Kafka).

Later, I would follow up with a PR for the Official Apache Kafka Native image.

Should I reference this issue in my first PR?

**Comment by cescoffier** [comment]
It could be another alternative, but probably not the default. Our dev services must start relatively quickly (less than 5s on a regular machine) and not consume too much memory. This is why we are not using the default Kafka Container from test containers; the experience was sub-optimal. 

If the user needs to test against a Kafka config closer to prod, Test Resources can be used to start their own container.

**Comment by sandonjacobs** [comment]
Agreed... Having both options is feasible. I'm pretty sure the Official Kafka Native image would meet the needs and the Official Kafka Image starts much faster than previous images, especially since the removal of ZK.

**Comment by cescoffier** [comment]
That's right! When we tested that, we needed zookeeper. That's not the case anymore.

Let's add more options and see how they behave.

**Comment by ozangunalp** [comment]
Resolved with #53395
